pydrive2/auth.py

```diff
@@ -308,18 +309,24 @@ def ServiceAuth(self):
         if set(self.SERVICE_CONFIGS_LIST) - set(self.client_config):
             self.LoadServiceConfigSettings()
         scopes = scopes_to_string(self.settings["oauth_scope"])
-        client_service_json = self.client_config.get("client_json_file_path")
-        creds_dict = self.client_config.get("client_creds_dict")
-        if creds_dict:
+        keyfile_name = self.client_config.get("client_json_file_path")
+        keyfile_dict = self.client_config.get("client_json_dict")
+        keyfile_json = self.client_config.get("client_json")
+
+        if not keyfile_dict and keyfile_json:
+            # Compensating for missing ServiceAccountCredentials.from_json_keyfile
+            keyfile_dict = json.loads(keyfile_json)
```

Comment:
Note, `from_json_keyfile_name` is naively loading the json too and any errors pass right through. Same here.